awswrangler/catalog/_utils.py

```diff
@@ -124,14 +125,35 @@ def sanitize_column_name(column: str) -> str:
     return _sanitize_name(name=column)
 
 
-def sanitize_dataframe_columns_names(df: pd.DataFrame) -> pd.DataFrame:
-    """Normalize all columns names to be compatible with Amazon Athena and the AWS Glue Catalog.
+def rename_duplicate_columns(df: pd.DataFrame) -> pd.DataFrame:
+    """Append an incremental number to duplicate column names to conform with Amazon Athena.
+    
+    Also handles potential new duplicated conflicts by appending another `_n`
+    to the end of the column name if it conflicts.
+
+    >>> df_rename = wr.catalog.rename_duplicate_columns(df=pd.DataFrame({'A': [1, 2], 'a': [3, 4], 'a_1': [4, 6]}))
+    """
+    names = df.columns
```

Comment:
Have you considered these solutions in [SO](https://stackoverflow.com/questions/40774787/renaming-columns-in-a-pandas-dataframe-with-duplicate-column-names) which seem more elegant and less memory intensive?

Reply:
Oh, that's nice, I like that solution a lot.